### PR TITLE
Add support for FIDO2 WebAuthn MFA

### PR DESCRIPTION
**Pull request by ALescouet**
Add Okta FIDO2 WebAuthn workflow as a possible second factor using the fido2 python library.

Use --factor-priority webauthn to change this new factor priority.

Signed-off-by: Alexis Lescouet <alexis@lescouet.com>

### CODE REVIEW

**Comment by ALescouet** [comment]
@zdave What are your thoughts on this PR? I couldn't test with other hardware tokens but it works with the Yubikey 5, I also own an older yubikey 4 that I could test with if you feel that is needed.

**Comment by zdave** [comment]
I don't have access to a GlobalProtect+Okta VPN any more, so I can't really test anything. I'd be happy to submit this if it's working for you though. A few comments:

- If you're going to use `click.echo` and `click.prompt` it would be good to convert the existing `print`/`input` calls to use these too.
- I would make the `fido2` import optional, like `pyotp` is, so people don't need to install it unless they want to use it.
- There's quite a bit of FIDO2 MFA logic so I would pull it out into its own function.
- The `websafe_decode(websafe_encode(obj))` bit in the `b64` function seems like a no-op. If it isn't, should add a comment explaining what is going on!

File: openconnect-gp-okta [hunk 1]
```diff
@@ -63,6 +87,60 @@ def okta_auth(s, domain, username, password, factor_priorities, totp_key):
...
+                    click.echo("Please insert a suitable device if you wish to continue with webauthn MFA.")
+                    if click.confirm("Continue with webauthn MFA?"):
+                        device = next(list_devices(), None)
+                    else:
+                        break
+
+                if device is None:
+                    click.echo("Falling back to other MFA method.")
+                    url = r['_links']['prev']['href']
+                    post_json(s, url, {'stateToken': r['stateToken']})
```
Comment (zdave) [inline openconnect-gp-okta]: Should this assign to `r`? Or should we keep the original `r` around? Seems wrong to continue with the one from `webauthn/verify`...

Comment (ALescouet) [reply openconnect-gp-okta]: Thanks for pointing that out, it was an indeed a mistake. This should be fixed in the refactored version.

**Comment by ALescouet** [comment]
@zdave Thank you for reviewing this PR. I think I addressed all your comments, please tell me if anything else needs modifying.

**Pull request closed without merging**